### PR TITLE
Feat: Support list of masks in `AugmentationSequential`

### PR DESCRIPTION
**Pull request by johnnv1**
#### Changes
This PR put the changes from https://github.com/kornia/kornia/pull/2421 by @ashnair1 into a single commit based on the main branch

Fixes https://github.com/kornia/kornia/discussions/2417

@shijianjian can you help me figure out what tests we need to add here?

#### Type of change

- [ ] 📚  Documentation Update
- [x] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

### CODE REVIEW

**Review by edgarriba — APPROVED** [review]

File: kornia/augmentation/container/ops.py [hunk 1]
```diff
@@ -260,6 +266,69 @@ def transform(cls, input: Tensor, module: Module, param: ParamItem, extra_args:
...
+                tfm_input.append(tfm_inp)
+                input = tfm_input
+
+        elif isinstance(module, K.container.ImageSequentialBase):
+            tfm_input = []
+            seq_params = cls.get_sequential_module_param(param)
+            for inp in input:
+                tfm_inp = module.transform_masks(inp, params=seq_params, extra_args=extra_args)
+                tfm_input.append(tfm_inp)
+            input = tfm_input
```
Comment (johnnv1) [inline kornia/augmentation/container/ops.py:320]:
```suggestion
        elif isinstance(module, K.ImageSequential) and not module.is_intensity_only():
            tfm_input = []
            seq_params = cls.get_sequential_module_param(param)
            for inp in input:
                tfm_inp = module.transform_masks(inp, params=seq_params, extra_args=extra_args)
                tfm_input.append(tfm_inp)
                input = tfm_input

        elif isinstance(module, K.container.ImageSequentialBase):
            tfm_input = []
            seq_params = cls.get_sequential_module_param(param)
            for inp in input:
                tfm_inp = module.transform_masks(inp, params=seq_params, extra_args=extra_args)
                tfm_input.append(tfm_inp)
            input = tfm_input
```

@shijianjian any clue why this is falling just when running as `float64`? 🤔

Comment (shijianjian) [reply kornia/augmentation/container/ops.py:320]: weird, I checked the code, and nowhere mentioned float32 only.

Comment (johnnv1) [reply kornia/augmentation/container/ops.py:320]: yep, I don't understand why it's crashing the shape of the data with fp64. Maybe, because randomness isn't running when we don't put fp64. 🤔

File: tests/augmentation/test_container.py [hunk 2]
```diff
@@ -672,6 +673,39 @@ def test_inverse_and_forward_return_transform(self, random_apply, device, dtype)
         if random_apply is False:
             reproducibility_test((inp, mask, bbox, keypoints, bbox_2, bbox_wh, bbox_wh_2), aug)
 
+    def test_transform_list_of_masks_and_boxes(self, device, dtype):
+        input = torch.randn(2, 3, 256, 256, device=device, dtype=dtype)
+        mask = [
+            torch.ones(1, 3, 256, 256, device=device, dtype=dtype),
```
Comment (johnnv1) [inline tests/augmentation/test_container.py:679]: which failed randomly before, is sometimes, one of the augmentations complains about the masks be not batched... but it looks to be fine with a list of batched masks... 

@shijianjian Do you think it is fine to go this way? Looks like it partially adds the support to list of masks

Comment (shijianjian) [reply tests/augmentation/test_container.py:679]: Great. LGTM.

Comment (shijianjian) [reply tests/augmentation/test_container.py:679]: BTW, maybe not in this PR. I think we also wanted to support instance segmentation. So, we will need something similar to bboxes later on.

Comment (johnnv1) [reply tests/augmentation/test_container.py:679]: I don't get what you mean, because from what I see, this patch already supports a list of bbox
